### PR TITLE
[MIST-1068] Move queryStarter and queryRemover from Group to ApplicationInfo

### LINKED ISSUE #1068: Move queryStarter and queryRemover from Group to ApplicationInfo
`ApplicationInfo` needs to contain jar files and app information. We need to move `queryStarter `and `queryRemover` from `Group` to `ApplicationInfo`.

Previous PR: https://github.com/snuspl/mist/commit/44460db1b4c748a82d7dd820fbff044fe41a1524

### PR DESCRIPTION
**Pull request by kyungissac**
The PR addresses #1068.

Closes #1068.

### CODE REVIEW

File: mist-core/src/main/java/edu/snu/mist/core/task/groupaware/GroupAwareQueryManagerImpl.java [hunk 1]
```diff
@@ -208,7 +208,7 @@ public Query createAndStartQuery(final String queryId,
     // Start the submitted dag
     // TODO : [MIST-1016] get the appropriate Group for this applicationInfo.
     final Group group = applicationInfo.getRandomGroup();
```
Comment (taegeonum) [inline mist-core/src/main/java/edu/snu/mist/core/task/groupaware/GroupAwareQueryManagerImpl.java]: please remove this line

**Comment by kyungissac** [comment]
@taegeonum I've addressed your comment. Thanks!

File: mist-core/src/main/java/edu/snu/mist/core/task/checkpointing/DefaultCheckpointManagerImpl.java [hunk 2]
```diff
@@ -178,7 +178,7 @@ public void deleteGroup(final String groupId) {
           new Object[] {groupId});
       return;
     }
-    group.getQueryRemover().deleteAllQueries();
+    group.getApplicationInfo().getQueryRemover().deleteAllQueries();
```
Comment (taegeonum) [inline mist-core/src/main/java/edu/snu/mist/core/task/checkpointing/DefaultCheckpointManagerImpl.java]: This will delete all queries in the application.

**Comment by kyungissac** [comment]
@taegeonum I've addressed your comment. Thanks!

Due to this change, the method `deleteAllQueries()` in the `QueryRemover` interface is currently not used anywhere. Do you think we should remove it, or should we keep it in case we delete an app?

**Comment by taegeonum** [comment]
@kyungissac I think it would be good to keep the interface now.  I will merge this PR. Thanks!

**Pull request merged**